src/socket_utilities/request_protocol/request_protocol.c

```diff
@@ -227,7 +227,13 @@ QueryResponse* parse_query_response(const int client_fd, ErrorInfo *err)
 }
 
 void get_message_size(const int client_fd, uint32_t* message_size, ErrorInfo *err) {
-    if (read(client_fd, message_size, sizeof(uint32_t)) <= 0) {
+    const ssize_t bytes = read(client_fd, message_size, sizeof(uint32_t));
+    if (bytes == 0) {
```

Comment:
tu nie bedzie moze -1 jak jakis blad? Tylko sie upewniam bo moze rzeczywiscie jest tak ze na nie blokujacym zwraca 0 jak jest rozloaczony czy cos

Reply:
Testowałam i zazwyczaj jak zatrzymywałam controller to albo zwracało -1 i ustawiało errno = EAGAIN w kółko albo 0 i brak błędu. 

Wydaje mi się, że jak wszystko inne poprawnie działa to nie ma opcji, że próba odczytu socketu controllera zwraca 0 a ten controller nadal działa (bo select powiedział że ten controller jest gotowy do odczytu, czyli coś nam wysłał, nigdzie indziej tego nie odczytaliśmy więc powinno tam być).  

Nie mogę tego znaleźć o read ale w man page dla recv (który z tego co rozumiem jest readem tylko do socketów) jest

       When a stream socket peer has performed an orderly shutdown, the
       return value will be 0 (the traditional "end-of-file" return).
       
Podsumowując moja 3-argumentową rozprawkę założyłam, że read 0 = zamknięty socket. 
Ale mogę zmienić na recv dla pewności, co myślisz?

Reply:
git